### PR TITLE
Proto autodoc

### PR DESCRIPTION
**Pull request by tgaugry**
First try at the doc autogeneration. Include a .gitignore & some doc as a bonus.
Example of generated file [Simulation_doc.txt](https://github.com/SofaDefrost/plugin.SofaPython3/files/3228593/Simulation_doc.txt)

Usage shown in Binding_Simulation.cpp

TODO:
- DONE ~Integrate doc/mkdoc.sh in the CMakeLists.txt~ 
- Deduce paths of inputs &co -> moved to trello
- Deduce paths of headers for propagation  -> moved to trello

### CODE REVIEW

**Comment by damienmarchal** [comment]
hi thierry,

Thanks for the PR.  Isn't Simulation_doc.h missing ?

**Comment by tgaugry** [comment]
`Simulation_doc.h` has to be generated. An example of it is in the original comment.

**Comment by damienmarchal** [comment]
Nice.

File: doc/mkdoc.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/bin/sh
+
+sofa_src=/home/tgaugry/sofa/src/master
+build_dir=/home/tgaugry/sofa/build/master/Release
+sofapython3_src=$(realpath ..)
+include_dirs="-I/usr/lib64/llvm5.0/lib/clang/5.0.1/include -I${build_dir}/include -I${sofa_src}/SofaKernel/framework -I${sofapython3_src}/binding -I/usr/include/python3.7m"
```
Comment (damienmarchal) [inline doc/mkdoc.sh]: Bheu...c'est quoi ça :)

Comment (tgaugry) [reply doc/mkdoc.sh]: That's actually in the TODO in the first message :D I wanted to be sure we take this approach before committing to cmake madness

**Comment by tgaugry** [comment]
Ok, so it does work with `Binding_Simulation.h`. As for the other files... Code is commented for now.
Problem now is that I have no way of finding the original file (`Simulation.h`).
Paths do not have the same structure on both sides. For example, in sofa we get `SofaKernel/framework/sofa/core/objectmodel/Base.h` and in SofaPython3 we get `bindings/Sofa/src/SofaPython3/Sofa/Core/Binding_Base.h`.
We could glob recurse our way through all of it, that's kinda ugly. Any better solution ?

**Comment by damienmarchal** [comment]
Hello Thierry,

With the autogenerated files who do we tune the docs to make really present the python binding (if it extends the c++ one)

Eg: 

static const char *__doc_binding_Simulation_init = R"doc(
   This is a function that do XXX::YYY::ZZZ
   
   More details and bli bla bla.
       
   Signature:
        :param float: xx
        :param int: yy 

   Example of use:
        import SofaSimulation.runTime()
)doc";


From my inderstanding...line 1 and other is what mkdocs is doing. pybind11 was doint the signature..and in any case we will need the example of use. 

Do you have any idea how could we do all that ?

**Comment by tgaugry** [comment]
I don't know if this is actually needed. This is a binding, not an abstraction layer. For each function, usage should be the same in both C++ and python. The only difference I see would be the `import` statement, which can go in a separated doc.

The second solution would be maybe to merge the automated doc with an example, like 
```.def("reset", &Simulation::reset, "node"_a, std::string(__doc_sofa_simulation_Simulation_reset) + " Example of use : ..."); ```

**Comment by damienmarchal** [comment]
But, because of the pythonic way of doing things, the python binding we are currently writing is closer to an abstraction layer than a pure binding so to me it is crucial we keep the freedom to write the docstrings.

File: README.md [hunk 2]
```diff
@@ -1,18 +1,18 @@
 # plugin.SofaPython3
 A python plugin for Sofa offering a pythonic interface and python3 support.
 
-
-# Requirement Install
+## Install Requirements
 - pybind11 (minimal 2.2.4)
 - cmake (minimal 3.12)
 - developement package for python3 (-dev)
+- python-ad (for tests)
```
Comment (damienmarchal) [inline README.md]: non non, il faut virer les lignes avec ad.

**Comment by marques-bruno** [comment]
fixed in #16

**Pull request closed without merging**